service/service.go

```diff
@@ -513,6 +513,22 @@ func (s *Service) GetToken(ctx context.Context, req *pbsvc.UserRequest) (*pbsvc.
 		return nil, status.Error(codes.Unauthenticated, err.Error())
 	}
 
+	// lookup user in token table for existing user
+	existingToken, err := getExistingToken(retrievedUser.GetUuid())
+	if err != nil {
+		logger.Error(consts.GetAuthTokenTag, consts.MsgErrGetExistingToken, err.Error())
+	}
+	if existingToken != nil && existingToken.permission == retrievedUser.GetPermissionLevel() {
```

Comment:
If you made `getExistingToken` return an `error` on the last `return statement`, then you would not need the condition `existingToken != nil`

Reply:
i pushed changes, not sure if the change is what you meant